scipy/linalg/decomp.py

```diff
@@ -371,126 +415,170 @@ def eigh(a, b=None, lower=True, eigvals_only=False, overwrite_a=False,
     True
 
     """
+    # set lower
+    uplo = 'L' if lower else 'U'
+    # Set job for Fortran routines
+    _job = (eigvals_only and 'N') or 'V'
+
+    drv_str = [None, "ev", "evd", "evr", "evx", "gv", "gvd", "gvx"]
+    if driver not in drv_str:
+        raise ValueError('"{}" is unknown. Possible values are "None", "{}".'
+                         ''.format(driver, '", "'.join(drv_str[1:])))
+
     a1 = _asarray_validated(a, check_finite=check_finite)
     if len(a1.shape) != 2 or a1.shape[0] != a1.shape[1]:
-        raise ValueError('expected square matrix')
+        raise ValueError('expected square "a" matrix')
     overwrite_a = overwrite_a or (_datacopied(a1, a))
-    if iscomplexobj(a1):
-        cplx = True
-    else:
-        cplx = False
+    cplx = True if iscomplexobj(a1) else False
+    n = a1.shape[0]
+    drv_args = {'overwrite_a': overwrite_a}
+
     if b is not None:
         b1 = _asarray_validated(b, check_finite=check_finite)
         overwrite_b = overwrite_b or _datacopied(b1, b)
         if len(b1.shape) != 2 or b1.shape[0] != b1.shape[1]:
-            raise ValueError('expected square matrix')
+            raise ValueError('expected square "b" matrix')
 
         if b1.shape != a1.shape:
-            raise ValueError("wrong b dimensions %s, should "
-                             "be %s" % (str(b1.shape), str(a1.shape)))
-        if iscomplexobj(b1):
-            cplx = True
-        else:
-            cplx = cplx or False
-    else:
-        b1 = None
+            raise ValueError("wrong b dimensions {}, should "
+                             "be {}".format(b1.shape, a1.shape))
 
-    # Set job for Fortran routines
-    _job = (eigvals_only and 'N') or 'V'
+        cplx = True if iscomplexobj(b1) else (cplx or False)
+        drv_args.update({'overwrite_b': overwrite_b})
 
-    # port eigenvalue range from Python to Fortran convention
-    if eigvals is not None:
-        lo, hi = eigvals
-        if lo < 0 or hi >= a1.shape[0]:
-            raise ValueError('The eigenvalue range specified is not valid.\n'
-                             'Valid range is [%s,%s]' % (0, a1.shape[0]-1))
-        lo += 1
-        hi += 1
-        eigvals = (lo, hi)
+    # backwards-compatibility handling
+    subset_by_index = subset_by_index if (eigvals is None) else eigvals
 
-    # set lower
-    if lower:
-        uplo = 'L'
-    else:
-        uplo = 'U'
+    subset = (subset_by_index is not None) or (subset_by_value is not None)
+
+    # Both subsets can't be given
+    if subset_by_index and subset_by_value:
+        raise ValueError('Either index or value subset can be requested.')
+
+    # Take turbo into account if all conditions are met otherwise ignore
+    if turbo:
+        if b is not None:
+            driver = 'gvx' if subset else 'gvd'
+
+    # Check indices if given
+    if subset_by_index:
+        lo, hi = [int(x) for x in subset_by_index]
```

Comment:
Sorry, I can't help myself : )
```suggestion
        lo, hi = (int(x) for x in subset_by_index)
```
I figure I should show that I'm reading this, right?
Feel free to ignore.

Reply:
😃